### PR TITLE
(#164) - test minimumForPouchDB mode

### PR DESCRIPTION
**Pull request by marten-de-vries**
Won't pass Travis until pouchdb/pouchdb#3363 is merged.

### CODE REVIEW

**Comment by marten-de-vries** [comment]
No idea how to fix Travis for this, I can't reproduce the npm error locally. Local testing procedure: https://gist.github.com/marten-de-vries/bd0013875b5e53675f56

If anyone's got a clue on what's happening, I'd love to hear it.

File: .travis.yml [hunk 1]
```diff
@@ -14,9 +17,14 @@ before_script:
...
   - "sudo sed -i 's/^127\\.0\\.0\\.1.*$/127.0.0.1 localhost/' /etc/hosts"
+
+  - "cd node_modules/pouchdb && npm install && cd -"
+  - "rm -fr node_modules/pouchdb/node_modules/express-pouchdb"
+  - "ln -s ../../express-pouchdb node_modules/pouchdb/node_modules/express-pouchdb"
+
   - "cd node_modules/pouchdb-server && npm install && cd -"
   - "rm -fr node_modules/pouchdb-server/node_modules/express-pouchdb"
-  - "ln -s ../../.. node_modules/pouchdb-server/node_modules/express-pouchdb"
+  - "ln -s ../../express-pouchdb node_modules/pouchdb-server/node_modules/express-pouchdb"
```
Comment (nolanlawson) [inline .travis.yml]: I'm guessing that the above 6 lines don't actually need to be changed, and that that's why it's failing in Travis? TBH I just messed around with this until it worked; it's a pretty hacky solution.

Comment (marten-de-vries) [inline .travis.yml]: I tried pretty much all variations I could up with, including copying over what it was and just replacing pouchdb-server with pouchdb (that was the first try). The current commit is just another experiment that failed.

**Comment by nolanlawson** [comment]
So it looks like the issue is that express-pouchdb is trying to run its dependency (pouchdb) who has its own dependency on express-pouchdb, hence the circular problem.

**Comment by nolanlawson** [comment]
Actually wait, the dep graph is pouchdb -> pouchdb-server -> express-pouchdb, so you need to swap out the sub-sub-dependency, not add a direct express-pouchdb dependency to pouchdb.

**Comment by marten-de-vries** [comment]
Experimented some more, the current Travis config seems to work! :)

**Comment by nolanlawson** [comment]
Cool, an enthusiastic +1!

**Comment by nolanlawson** [comment]
A few things to possibly worry about, though:

1. `minimum` takes 6 minutes to run, whereas `full` takes 9. Should the difference be that great?
2. `minimum` has lots and lots of errors/warnings in the console. The tests still pass, but there are errors like:

```
Possibly unhandled Error: Can't set headers after they are sent.
    at ServerResponse.OutgoingMessage.setHeader (http.js:690:11)
    at Object.exports.sendJSON (/home/travis/build/pouchdb/express-pouchdb/lib/utils.js:185:9)
    at Object.exports.sendError (/home/travis/build/pouchdb/express-pouchdb/lib/utils.js:177:11)
    at /home/travis/build/pouchdb/express-pouchdb/lib/routes/compact.js:15:20
    at tryCatch1 (/home/travis/build/pouchdb/express-pouchdb/node_modules/pouchdb/node_modules/bluebird/js/main/util.js:63:19)
    at Promise$_callHandler [as _callHandler] (/home/travis/build/pouchdb/express-pouchdb/node_modules/pouchdb/node_modules/bluebird/js/main/promise.js:695:13)
    at Promise$_settlePromiseFromHandler [as _settlePromiseFromHandler] (/home/travis/build/pouchdb/express-pouchdb/node_modules/pouchdb/node_modules/bluebird/js/main/promise.js:711:18)
    at Promise$_settlePromiseAt [as _settlePromiseAt] (/home/travis/build/pouchdb/express-pouchdb/node_modules/pouchdb/node_modules/bluebird/js/main/promise.js:868:14)
    at Promise$_settlePromises [as _settlePromises] (/home/travis/build/pouchdb/express-pouchdb/node_modules/pouchdb/node_modules/bluebird/js/main/promise.js:1006:14)
    at Promise$_rejectPromises [as _rejectPromises] (/home/travis/build/pouchdb/express-pouchdb/node_modules/pouchdb/node_modules/bluebird/js/main/promise.js:999:10)
    at Async$_consumeFunctionBuffer [as _consumeFunctionBuffer] (/home/travis/build/pouchdb/express-pouchdb/node_modules/pouchdb/node_modules/bluebird/js/main/async.js:74:12)

Possibly unhandled Error: once called  more than once
    at /home/travis/build/pouchdb/express-pouchdb/node_modules/pouchdb/lib/utils.js:464:13
    at /home/travis/build/pouchdb/express-pouchdb/node_modules/pouchdb/node_modules/argsarray/index.js:14:18
    at tryCatch1 (/home/travis/build/pouchdb/express-pouchdb/node_modules/pouchdb/node_modules/bluebird/js/main/util.js:63:19)
    at Promise$_callHandler [as _callHandler] (/home/travis/build/pouchdb/express-pouchdb/node_modules/pouchdb/node_modules/bluebird/js/main/promise.js:695:13)
    at Promise$_settlePromiseFromHandler [as _settlePromiseFromHandler] (/home/travis/build/pouchdb/express-pouchdb/node_modules/pouchdb/node_modules/bluebird/js/main/promise.js:711:18)
```

**Comment by marten-de-vries** [comment]
@nolanlawson 
1) The time difference is valid: authentication, /_security & validate_doc_update all need to do db reads, often every request. Maybe the situation could be improved upon a bit by caching some stuff in memory, but in the end these features just come at a price.

2) Yes, the first is probably the compaction hack we discussed coming back to bite me. Currently I'm not convinced those are all minimum-only so I opened: #194.

**Comment by nolanlawson** [comment]
Yep, for #1, it sounds like it would be a good idea to implement caching for those plugins, as we did for pouchdb-all-dbs.

**Comment by marten-de-vries** [comment]
All those plug-ins will need a maximum cache size to prevent using too much memory; apart from that optimizing them further all boiles down to caching recently used docs.

That's probably easiest to do either in PouchDB core or as a separate plug-in (pouchdb-cache could keep allDocs()'s output in memory if it's not too large & also the x most recently used docs, again keeping in mind memory constraints).

**Comment by nolanlawson** [comment]
Interesting. I don't think it's appropriate for pouch core, but yeah, that could be awesome as a plugin based on pouchdb-wrapper. Think of the perf benefits for map/reduce. :)

**Pull request merged**